app/src/main/java/com/google/android/samples/socialite/ui/videoedit/VideoEditScreen.kt

```diff
@@ -104,6 +108,19 @@ fun VideoEditScreen(
 
     val isProcessing = viewModel.isProcessing.collectAsState()
 
+    LaunchedEffect(lifecycleOwner, context) {
+        viewModel.videoSaveState.collect { state ->
+            when (state) {
+                VideoSaveState.VIDEO_SAVE_SUCCESS ->
+                    Toast.makeText(context, "Edited video saved", Toast.LENGTH_LONG).show()
+                VideoSaveState.VIDEO_SAVE_FAIL ->
+                    Toast.makeText(context, "Error applying edits on video", Toast.LENGTH_LONG)
+                        .show()
+                VideoSaveState.PENDING -> Unit
+            }
+        }
+    }
```

Comment:
Should add `viewModel.markMessageShown()` to avoid showing toast twice, for example: onStop and onStart again

Reply:
https://developer.android.com/topic/architecture/ui-layer/events#consuming-trigger-updates

Reply:
@hoc081098 
I think it's better to change to the existing [collectAsStateWithLifecycle](https://medium.com/androiddevelopers/consuming-flows-safely-in-jetpack-compose-cde014d0d5a3)

Reply:
@yongsuk44 nice, but `LifecycleResumeEffect` is more suitable for flow collection 🙏 

```kotlin
val scope = rememberCoroutineScope()
LifecycleResumeEffect(scope) {
  // ON_RESUME code is executed here
  val job = viewModel.videoSaveState
     .onEach{ /*handler*/}
     .launchIn(scope)

  onPauseOrDispose {
    // do any needed clean up here
    job.cancel()
  }
}
```

Reply:
@hoc081098
Thanks for the feedback, I have question.

1. "Why are you 'State Hoisting' the state in the ViewModel and then bringing it back to the Composable? 
It seems to go against the Unidirectional Data Flow (UDF) design pattern."

2. "Considering the link I provided, collectAsStateWithLifecycle is designed to simplify the task of activating Flow collection when the app is in the foreground, making it easier for developers to manage state. So, why use LifecycleResumeEffect?"

Reply:
collectAsStateWithLifecycle() = `produceState + repeatOnLifecycle + flow.collect{}`

-> we only collect the flow 👍, we don't need a returned Compose `State<T>`, don't need the recomposition of the `@Composable`.
`LifecycleResumeEffect `is enough

Reply:
@hoc081098 😀 I understand, thanks